tests/PHPStan/Analyser/LegacyNodeScopeResolverTest.php

```diff
@@ -8003,7 +8003,7 @@ public function dataPassedByReference(): array
 				'$arr',
 			],
 			[
-				'array{0?: string}',
+				'array<string>',
```

Comment:
did we loose precision in this test?

Reply:
Yes well the new sanity check I added before parsing catches it and it bails out. The grammar doesn't actually check for delimiters so it sees 'a' as a literal a and is happy with that.